community/index/src/main/java/org/neo4j/index/gbptree/TreeNode.java

```diff
@@ -392,7 +392,6 @@ private int readRecordsWithInsertRecordInPosition( PageCursor cursor, Consumer<P
         cursor.getBytes( into, 0, insertPosition * recordSize );
 
         // Read newRecord
-        // TODO: A bit expensive to wrap in a PageCursor tin foil just to write this middle key
         PageCursor buffer = ByteArrayPageCursor.wrap( into, insertPosition * recordSize, recordSize );
         newRecordWriter.accept( buffer );
```

Comment:
I would like to get rid of the temp copying all together in the end, by using directly moving stuff from cursor to cursor, or in case of plain insert, move around keys/values inside the cursor w/o buffer in between. I think we should defer this to a slightly later optimization stage.